### PR TITLE
Reorg serialization samples in TOC

### LINKED ISSUE #14447: Reorganize the serialization samples TOC in the .NET guide
Related to #12207 and https://github.com/dotnet/docs/pull/14208/files#r324394406

* Retire the link farm doc and redirect its URL to the top-level serialization overview doc.
* Put a Samples node under Binary and XML sections
* Divide up the samples into binary and XML, and add nodes under the appropriate Samples nodes for them.

When we have System.Text.Json samples we'll add a Samples node under the JSON section as well.

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #14447

### CODE REVIEW

File: docs/standard/serialization/toc.yml [hunk 1]
```diff
@@ -73,23 +75,12 @@
...
-      - name: Custom Serialization Order With XmlSerializer
-        href: custom-serialization-order-with-xmlserializer.md
-      - name: SchemaImporterExtension Technology Sample
-        href: schemaimporterextension-technology-sample.md
-      - name: Version Tolerant Serialization Technology Sample
-        href: version-tolerant-serialization-technology-sample.md
-      - name: Web Services Generics Serialization Technology Sample
-        href: web-services-generics-serialization-technology-sample.md
+    - name: Samples
+      href: serialization-samples-xml.md
```
Comment (mairaw) [inline docs/standard/serialization/toc.yml]: why remove the children articles from the TOC here?

File: docs/standard/serialization/serialization-samples-binary.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (mairaw) [inline docs/standard/serialization/serialization-samples-binary.md]: when I run SEO tests on CATS, it complains when we have duplicate words on the path like serialization here shows up twice.
I'm not going after fixing the migrated samples, but I try to follow that rule when creating new articles.

**Review by rpetrusha — APPROVED** [review]
This is a much more sensible organization of the samples, @tdykstra. You can merge when you're ready.

**Comment by mairaw** [comment]
Sorry I was late with my review 😢

File: docs/standard/serialization/toc.yml (hunk 1 above)
Comment (tdykstra) [reply docs/standard/serialization/toc.yml]: One of the samples is for both XML and binary.  If I kept them in the TOC I would have had to duplicate that node.

File: docs/standard/serialization/serialization-samples-binary.md (hunk 2 above)
Comment (tdykstra) [reply docs/standard/serialization/serialization-samples-binary.md]: These articles were created by adding -xml and -binary to the existing file name.  I can rename them in a separate PR before master goes to live.

**Pull request merged**